src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/NestedTypesShouldNotBeVisible.cs

```diff
@@ -50,12 +51,12 @@ public override void Initialize(AnalysisContext analysisContext)
             analysisContext.RegisterCompilationStartAction(
                 compilationStartContext =>
                 {
-                    Compilation compilation = compilationStartContext.Compilation;
```

Comment:
Why the changes to explicitly use `wellKnownTypeProvider` here, when the underlying implementation for `GetOrCreateTypeByMetadataName` already uses it?

Reply:
The idea was to avoid the overhead of the cache call.